tests/foreman/api/test_discoveryrule.py

```diff
@@ -177,13 +177,13 @@ def test_positive_multi_provision_with_rule_limit(
 
     :CaseImportance: High
     """
-    for _ in range(2):
-        discovered_host = module_target_sat.api_factory.create_discovered_host()
+    discovered_host1 = module_target_sat.api_factory.create_discovered_host()
+    discovered_host2 = module_target_sat.api_factory.create_discovered_host()
 
     rule = module_target_sat.api.DiscoveryRule(
         max_count=1,
         hostgroup=module_discovery_hostgroup,
-        search_=f'name = {discovered_host["name"]}',
+        search_=f'name = {discovered_host1["name"]}',
```

Comment:
This test is for multi_provision, so shouldn't we include both discovered hosts for provision and delete both in teardown?

Reply:
This test is for multi_provision with a rule limit, here we are only provisioning discovered hosts which match the created discovery rule(1 host in our test).

Reply:
You got it correct @shubhamsg199.